src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4592,6 +4592,41 @@ public override BoundNode VisitThrowExpression(BoundThrowExpression node)
             return result;
         }
 
+        public override BoundNode VisitYieldReturnStatement(BoundYieldReturnStatement node)
+        {
+            BoundExpression expr = node.Expression;
+            if (expr == null)
+            {
+                return null;
+            }
```

Comment:
>} [](start = 12, length = 1)

Please add a test where `expr.HasErrors` and also results in nullable warnings.

Reply:
Sorry, it looks like I had forgotten to push the commit updating tests. There are already tests for this.
Please take another look. 

I much prefer the behavior that avoids cascading diagnostics. We should discuss more. I filed a follow-up issue: https://github.com/dotnet/roslyn/issues/31121